src/table/pagination-table/components/head/TableHeadWrapper.tsx

```diff
@@ -19,8 +19,10 @@ function TableHeadWrapper({ areBasicFeaturesEnabled }: TableHeadWrapperProps) {
   const headRowRef = useRef<HTMLTableRowElement>(null);
 
   useEffect(() => {
-    headRowRef.current && setHeadRowHeight(headRowRef.current.getBoundingClientRect().height);
-  }, [headRowRef.current, styling.head.fontSize, headRowRef.current?.getBoundingClientRect().height]);
+    if (headRowRef.current) {
+      setHeadRowHeight(headRowRef.current.getBoundingClientRect().height);
+    }
+  }, [styling.head.fontSize, columnWidths, setHeadRowHeight]);
```

Comment:
There a couple of scenarios when this use effect needs to run:
- Column is re-sized
- Container element is re-sized
- Layout changes
- Theme change

The `columnWidths` will take care of the 3 first scenarios and the `styling.head.fontSize` the last.

Reply:
Not entirely true as columnWidths have a bug where it does not run it's effect every time it needs to. Will make a new PR for that fix.

Reply:
@LiKang6688  `setHeadRowHeight` is a React hook dispatch function. It does not change between renders so it will not triggers the `useEffect` on every render.

I added it, because otherwise eslint would complain about a missing dependency. You don't really need dispatch functions in the deps array but because it's coming from the `useContextSelector`eslint doesn't understand that. So had to add it to get rid of the warning.

Reply:
aha, I see. Thanks